arwen/cryptoapi/cryptoei.go

```diff
@@ -0,0 +1,266 @@
+package cryptoapi
+
+// // Declare the function signatures (see [cgo](https://golang.org/cmd/cgo/)).
+//
+// #include <stdlib.h>
+// typedef unsigned char uint8_t;
+// typedef int int32_t;
+//
+// extern int32_t sha256(void* context, int32_t dataOffset, int32_t length, int32_t resultOffset);
+// extern int32_t keccak256(void *context, int32_t dataOffset, int32_t length, int32_t resultOffset);
+// extern int32_t ripemd160(void *context, int32_t dataOffset, int32_t length, int32_t resultOffset);
+// extern int32_t verifyBLS(void *context, int32_t keyOffset, int32_t messageOffset, int32_t messageLength, int32_t sigOffset);
+// extern int32_t verifyEd25519(void *context, int32_t keyOffset, int32_t messageOffset, int32_t messageLength, int32_t sigOffset);
+// extern int32_t verifySecp256k1(void *context, int32_t keyOffset, int32_t keyLength, int32_t messageOffset, int32_t messageLength, int32_t sigOffset);
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/arwen"
+	"github.com/ElrondNetwork/arwen-wasm-vm/wasmer"
+)
+
+const blsPublicKeyLength = 96
+const blsSignatureLength = 48
+const ed25519PublicKeyLength = 32
+const ed25519SignatureLength = 64
+const secp256k1CompressedPublicKeyLength = 33
+const secp256k1UncompressedPublicKeyLength = 65
+const secp256k1SignatureLength = 64
```

Comment:
Optionally, you could capitalize these constants (`BLSPublicKeyLength`) to make them more conspicuous in the code below. They're potentially useful from the outside of the package as well.

Reply:
Done